cdap-data-fabric/src/main/java/co/cask/cdap/spi/data/sql/PostgresSqlStructuredTableAdmin.java

```diff
@@ -114,6 +122,20 @@ private boolean tableExistsInternal(Connection connection, StructuredTableId tab
     }
   }
 
+  private Set<String> getNonExistIndexes(Connection connection,
+                                         StructuredTableSpecification specification) throws SQLException {
+    DatabaseMetaData metaData = connection.getMetaData();
+    Set<String> existingIndexes = new HashSet<>();
+    Set<String> indexes = new HashSet<>(specification.getIndexes());
+    try (ResultSet rs = metaData.getIndexInfo(null, null,
+                                              specification.getTableId().getName(), false, false)) {
+      while (rs.next()) {
+        existingIndexes.add(rs.getString("COLUMN_NAME"));
```

Comment:
Also - `rs.getString("COLUMN_NAME")` can return null in certain [cases](https://docs.oracle.com/javase/7/docs/api/java/sql/DatabaseMetaData.html#getIndexInfo(java.lang.String,%20java.lang.String,%20java.lang.String,%20boolean,%20boolean))

Reply:
Yea, COLUMN_NAME will return the column name that is indexed, since the indexes in our spec is all column names, it is easy to compare if we use COLUMN_NAME. I don't think we create any of our indexes as type `tableIndexStatistic `